doc/supported_file_formats.adoc

```diff
@@ -40,6 +37,36 @@ Minetest is able to read several types of file formats for use as assets in-game
 
 Prior to 5.5.0, Minetest used to be able to load PCX, PPM, PSD, PVR, DDS, WAL, LMP and RGB as part of Irrlicht's image loading capabilities. These are no longer supported and should be converted to a format that is still supported.
 
+=== Optimizing images
+Normally when you export images with a regular image editor it is not exported in the most optimized way possible, and your editor will also embed metadata into the image. It is therefore standard practice to optimize your images in order to reduce their file size. This will not only reduce the size of your assets when a player downloads it but also reduce the time spent for media to be downloaded before joining a multiplayer server.
+
+==== PNG
+For losslessly optimizing PNG images, http://optipng.sourceforge.net/[optipng] is a good tool. It is a command-line program that is run from the command prompt or terminal.
+
+When run with no arguments optipng optimizes completely losslessly (no color, metadata or transparency is lost). However there are arguments in addition that can be added in order to further reduce the size or to preserve specific data you might want to keep. A more exhaustive list of arguments and what they do can be found in optipng's manual or in the help text available with `optipng -h`.
+
+[grid="all"]
+[options="header",stripes="even",cols="^.^,^.^,^.^"]
+|===
+| Argument
+| Description
+| Safe?
+
+| -strip all
+| Strips all image metadata your image editor has embedded into the image.
+| Yes (Image metadata is mostly useless for images used with Minetest)
+
+| -o7 -zm1-9
+| Sets both optipng and zlib's optimization/compression levels to the highest available. Will take significantly longer but can further reduce the file size.
+| Yes
+
+| -nc
+| Disables color type reduction. This is necessary if you have color data you want to preserve in transparent pixels, such as ones in leaf textures.
+| N/A
+|===
+
+A general-purpose script for optimizing images as aggressively as possible without losing color information can be found in Minetest Game: https://github.com/minetest/minetest_game/blob/master/utils/optimize_textures.sh[optimize_textures.sh]
+
```

Comment:
Info for TGA channels:

```
The TGA reader in Minetest supports several color channel configurations:
* 16 bit (1 bit alpha, 5 bit red, 5 bit green, 5 bit blue)
* 24 bit (8 bit blue, 8 bit green, 8 bit red)
* 32 bit (8 bit blue, 8 bit green, 8 bit red, 8 bit alpha)

Using 16 bit colors is an easy way to reduce filesize, with a slight loss of color fidelity.
Using 24 bit colors is the correct choice if no alpha channel is necessary.
```

Reply:
Info about TGA compression:

```
Minetest supports decoding run-length-encoded (RLE) TGA images, which store repeated pixels as one pixel and a repeat count. If your code only generates images, but does not need to edit them, it is easy to save those images as RLE compressed TGA.
```

Reply:
This might be a bit too much detail - I don't think low bit-depth TGAs should be endorsed just to shave off a few kb. If this much detail is added, there should also be info for PNG channels (both greyscale & low bit depth are supported I suppose).

Reply:
On the contrary, I think low bit-depth TGAs are an excellent for generated textures that do not need more than one alpha channel.

I agree that the info for PNG channels should be added. THere might be PNG feature combinations that the engine chokes on too, after all.